blocks/users-list/edit.js

```diff
@@ -1,3 +1,35 @@
-export default function Edit() {
-	return <div> Hello from edit </div>;
+import ServerSideRender from '@wordpress/server-side-render';
+import { useSelect } from '@wordpress/data';
+
+export default function Edit( { attributes, setAttributes } ) {
+	const userGroups = useSelect( ( select ) => {
+		return select( 'core' ).getEntityRecords( 'taxonomy', 'group' );
+	}, [] );
+
+	/* Display all group names */
+	const changeGroupName = ( role ) => {
+		if ( ! role ) {
+			setAttributes( { selectedValue: null } );
+			return;
+		}
+		const selectedValue = parseInt( role.target.value );
+		setAttributes( { selectedValue } );
+	};
+
+	return ( <div>
+		<label htmlFor="terms">Choose a role: </label>
+		<select name="terms" onBlur={ ( r )=>changeGroupName( r ) } id="terms">
+			<option value="">All</option>
+
+			{ userGroups && userGroups.map( function( taxo, i ) {
+				return <option key={ i } value={ taxo.id }>{ taxo.name }</option>;
```

Comment:
Should the `key` also be `taxo.id`?
```suggestion
				return <option key={ taxo.id } value={ taxo.id }>{ taxo.name }</option>;
```